rust/operator-binary/src/backend/mod.rs

```diff
@@ -96,6 +102,51 @@ pub struct SecretVolumeSelector {
     )]
     pub compat_tls_pkcs12_password: Option<String>,
 
+    /// An alternative name used for the TLS PKCS#12 keystore file.
+    ///
+    /// Has no effect if the `format` is not `tls-pkcs12`.
+    #[serde(
+        rename = "secrets.stackable.tech/format.tls-pkcs12.keystore-name",
+        default = "default_pkcs12_keystore_name"
+    )]
+    pub tls_pkcs12_keystore_name: String,
+
+    /// An alternative name used for the TLS PKCS#12 keystore file.
+    ///
+    /// Has no effect if the `format` is not `tls-pkcs12`.
+    #[serde(
+        rename = "secrets.stackable.tech/format.tls-pkcs12.truststore-name",
+        default = "default_pkcs12_truststore_name"
+    )]
+    pub tls_pkcs12_truststore_name: String,
+
+    /// An alternative name used for the TLS PEM certificate.
+    ///
+    /// Has no effect if the `format` is not `tls-pem`.
+    #[serde(
+        rename = "secrets.stackable.tech/format.tls-pem.cert-name",
+        default = "default_tls_pem_cert_name"
+    )]
+    pub tls_pem_cert_name: String,
+
+    /// An alternative name used for the TLS PEM certificate key.
+    ///
+    /// Has no effect if the `format` is not `tls-pem`.
+    #[serde(
+        rename = "secrets.stackable.tech/format.tls-pem.key-name",
+        default = "default_tls_pem_key_name"
+    )]
+    pub tls_pem_key_name: String,
+
+    /// An alternative name used for the TLS PEM certificate authority.
+    ///
+    /// Has no effect if the `format` is not `tls-pem`.
+    #[serde(
+        rename = "secrets.stackable.tech/format.tls-pem.ca-name",
+        default = "default_tls_pem_ca_name"
+    )]
+    pub tls_pem_ca_name: String,
```

Comment:
We could `#[serde(flatten)]` in `NamingOptions` instead to avoid this duplication.

Reply:
Done in 83fea67.

What do you think about using `#[serde(flatten)]` for `CompatibilityOptions` as well?

Reply:
That probably makes sense, yeah.

Reply:
Done in 94e78c5.

I also removed the `#[serde(deserialize_with = "SecretVolumeSelector::deserialize_some")]` argument because I _think_ it is not needed. I might be wrong tho.

Reply:
Please test both with/without the attribute before removing it - from what I remember, not having it caused some weird deserialization failures.

Reply:
Alright, good to know. I will add a bunch of unit tests for it.

Reply:
Added in 2d0a593.

Not providing the `secrets.stackable.tech/format.compatibility.tls-pkcs12.password` field will succeed deserialization without an error.